### PR TITLE
Deprecate Cluster Client #27242

### PR DESCRIPTION
**Pull request by helena**
#27242

### CODE REVIEW

File: akka-docs/src/main/paradox/cluster-client.md [hunk 1]
```diff
@@ -2,13 +2,14 @@
 
 @@@ note
 
+Cluster client has been deprecated in favor of [Akka gRPC](https://doc.akka.io/docs/akka-grpc/current/index.html).
+(FIXME https://github.com/akka/akka/issues/26175)
```
Comment (helena) [inline akka-docs/src/main/paradox/cluster-client.md]: Moved to above the typed note for prominence.

File: akka-docs/src/main/paradox/project/migration-guide-2.5.x-2.6.x.md [hunk 2]
```diff
@@ -195,6 +195,11 @@ Classic remoting is deprecated but can be used in `2.6.` Explicitly disable Arte
 specific to classic remoting needs to be moved to `akka.remote.classic`. To see which configuration options
 are specific to classic search for them in: [`akka-remote/reference.conf`](/akka-remote/src/main/resources/reference.conf)
 
+### Cluster client
+
+Cluster client has been deprecated as of 2.6 in favor of [Akka gRPC](https://doc.akka.io/docs/akka-grpc/current/index.html).
+It is not advised to build new applications with Cluster client, and existing users should migrate (FIXME https://github.com/akka/akka/issues/26175) to Akka gRPC.
+
```
Comment (helena) [inline akka-docs/src/main/paradox/project/migration-guide-2.5.x-2.6.x.md]: The FIXME is the actual migration example and recommendation @patriknw started in https://github.com/patriknw/akka-cluster-client-grpc/pull/2 needing reviews. This is just the entry of the change. When the samples are merged it should be added as links here.

File: akka-cluster-tools/src/main/scala/akka/cluster/client/ClusterClient.scala [hunk 3]
```diff
@@ -344,6 +346,7 @@ object ClusterClient {
  * Note that this is a best effort implementation: messages can always be lost due to the distributed
  * nature of the actors involved.
  */
+@deprecated("Use Akka gRPC instead.", since = "2.6.0")
```
Comment (helena) [inline akka-cluster-tools/src/main/scala/akka/cluster/client/ClusterClient.scala]: Is this the only place we'd need it though, on the object it of course breaks the build.

**Comment by chbatey** [comment]
I think the migration guide needs to come first

**Comment by helena** [comment]
@chbatey we can't get the migration sample in and add its links until maybe mid-august so I'm wondering about where to put this then in the sprint(s), see https://github.com/akka/akka/issues/26175#issuecomment-507024091

**Comment by helena** [comment]
Note PR for moving sample and guide https://github.com/akka/akka-samples/pull/116

File: akka-cluster-tools/src/main/scala/akka/cluster/client/ClusterClient.scala (hunk 3 above)
Comment (jrudolph) [reply akka-cluster-tools/src/main/scala/akka/cluster/client/ClusterClient.scala]: Is that ever used by users, though? It seems we should deprecate the things that users use, i.e. that are shown on the docs page. I'd suggest these:

 * `object ClusterClient` and/or `ClusterClient.props`
 * `object ClusterClientReceptionist`

File: akka-docs/src/main/paradox/cluster-client.md [hunk 4]
```diff
@@ -1,14 +1,19 @@
 # Classic Cluster Client
 
+@@@ warning
+
+Cluster Client is deprecated in favor of using [Akka gRPC](https://doc.akka.io/docs/akka-grpc/current/index.html).
```
Comment (jrudolph) [inline akka-docs/src/main/paradox/cluster-client.md]: Maybe also link to that paragraph: https://doc.akka.io/docs/akka-grpc/current/whygrpc.html#grpc-vs-akka-remoting that explains the trade-offs.

File: akka-docs/src/main/paradox/cluster-client.md [hunk 5]
```diff
@@ -227,3 +232,22 @@ This can be useful when initial contacts are provided from some kind of service
...
+We recommend using [Akka gRPC](https://doc.akka.io/docs/akka-grpc/current/index.html) over using Cluster Client.
+Before reading the migration guide it is advised to understand Akka gRPC. 
+
+### Migration steps
+
+The migration sample illustrates an approach to migrate from the deprecated Cluster Client to Akka gRPC, with minimal changes
+to your existing code. It is meant to be further adjusted to your requirements and use case
+(FIXME #26175 add link for scala sample, currently in PR, and a java sample).
+
+#### Cluster Client functions in Akka gRPC:
```
Comment (jrudolph) [inline akka-docs/src/main/paradox/cluster-client.md]: I guess this section will be fleshed out?

**Review by jrudolph — COMMENTED** [review]
A few comments

**Review by johanandren — APPROVED** [review]
LGTM with the same reservation as Johannes about where to deprecate

**Comment by patriknw** [comment]
Continued in https://github.com/akka/akka/pull/27609

**Pull request closed without merging**